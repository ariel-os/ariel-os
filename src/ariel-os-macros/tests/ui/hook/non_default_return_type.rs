#![no_main]

// FAIL: specifies a non-default return type
#[ariel_os::hook(usb_builder)]
fn usb_builder(builder: ariel_os::usb::UsbBuilder) -> usize {}
