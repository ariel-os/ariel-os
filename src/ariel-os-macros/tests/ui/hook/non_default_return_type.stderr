error: custom attribute panicked
 --> tests/ui/hook/non_default_return_type.rs:4:1
  |
4 | #[ariel_os::hook(usb_builder)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: message: hook functions must not have a non-default return type
